widgets/lib/progress_bar/curved_bar/woi_curved_bar.dart

```diff
@@ -0,0 +1,215 @@
+
+    /// Width of the border in case the arcBorders flag is set to true.
+    this.borderWidth = 5,
+
+    /// Length of the arc. By default the value is -20.
+    /// Arc length can be betweeen -20 and 90, inclusive. Hot restart is recommended when arc length is changed.
+    this.arcLength = -20,
+
+    /// Changes the direction of the arc. By default the direction is up.
+    this.arcDirection = ArcDirection.up,
+
+    /// Boolean value to determine if the center widget needs to rotate when rotating the arc.
+    /// By default the value is false.
+    this.rotateCenter = false,
+    super.key,
+  }) : assert(currentValue <= finalValue && currentValue >= 0,
+            'Current value cannot be greater than final value or less than 0');
+
+  final int finalValue;
+  final int currentValue;
+  final Widget center;
+  final double size;
+  final Color barColor;
+  final Color fillColor;
+  final Color backgroundColor;
+  final double backgroundBorderRadius;
+  final EdgeInsetsGeometry padding;
+  final double arcWidth;
+  final bool arcBorders;
+  final Color borderColor;
+  final double borderWidth;
+  final double arcLength;
+  final ArcDirection arcDirection;
+  final bool rotateCenter;
+
+  @override
+  State<WOICurvedBar> createState() => _WOICurvedBarState();
+}
+
+class _WOICurvedBarState extends State<WOICurvedBar> {
+  double arcLength = -20;
+
+  @override
+  void initState() {
+    if (widget.arcLength < -20) {
+      arcLength = -20;
+    } else if (widget.arcLength > 70) {
+      arcLength = 70;
+    } else {
+      arcLength = widget.arcLength;
+    }
+    super.initState();
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Container(
+      padding: widget.padding,
+      height: widget.size,
+      width: widget.size,
+      decoration: BoxDecoration(
+        color: widget.backgroundColor,
+        borderRadius: BorderRadius.circular(widget.backgroundBorderRadius),
+      ),
+      child: RotatedBox(
+        quarterTurns: widget.arcDirection == ArcDirection.up
+            ? 0
+            : widget.arcDirection == ArcDirection.right
+                ? 1
+                : widget.arcDirection == ArcDirection.left
+                    ? 3
+                    : widget.arcDirection == ArcDirection.down
+                        ? 2
+                        : 0,
+        child: CustomPaint(
+          painter: ArcPainter(
+            finalValue: widget.finalValue,
+            currentValue: widget.currentValue,
+            barColor: widget.barColor,
+            fillColor: widget.fillColor,
+            arcWidth: widget.arcWidth,
+            borders: widget.arcBorders,
+            borderColor: widget.borderColor,
+            borderWidth: widget.borderWidth,
+            arcLength: arcLength,
+          ),
+          child: widget.rotateCenter
+              ? widget.center
+              : RotatedBox(
+                  quarterTurns: widget.arcDirection == ArcDirection.up
```

Comment:
This isn't readable enough consider moving this to a method and using enums to return integer for better visibility. 

`int arcDirectionToQuarterTurns(ArcDirection direction) {
  switch (direction) {
    case ArcDirection.left:
      return 3;
    case ArcDirection.right:
      return 1;
    case ArcDirection.up:
      return 0;
    case ArcDirection.down:
      return 2;
  }
}`